docs/content/tutorial/ingress-tutorial.md

```diff
@@ -0,0 +1,139 @@
+
+### Setup an Ingress Controller
+
+First thing we will need is an ingress controller and we will use Nginx ingress controller in this tutorial. Based on your setup you can choose [one of the ways to install Nginx ingress controller from here](https://kubernetes.github.io/ingress-nginx/deploy/). This setup should work with other ingress controllers also but has not been tested.
+
+Once ingress controller is installed properly, let's verify that the installation succeeded (In my case all components of ingress controller were installed in ingress-nginx namespace):
+
+```
+kubectl get all -n ingress-nginx
+NAME                                           READY     STATUS    RESTARTS   AGE
+po/default-http-backend-66b447d9cf-4q8f7       1/1       Running   0          19d
+po/nginx-ingress-controller-58fcfdc6fd-2cwts   1/1       Running   0          19d
+
+NAME                       CLUSTER-IP      EXTERNAL-IP      PORT(S)                      AGE
+svc/default-http-backend   10.11.243.109   <none>           80/TCP                       19d
+svc/ingress-nginx          10.11.245.254   35.200.150.175   80:31000/TCP,443:30666/TCP   19d
+
+NAME                              DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+deploy/default-http-backend       1         1         1            1           19d
+deploy/nginx-ingress-controller   1         1         1            1           19d
+
+NAME                                     DESIRED   CURRENT   READY     AGE
+rs/default-http-backend-66b447d9cf       1         1         1         19d
+rs/nginx-ingress-controller-58fcfdc6fd   1         1         1         19d
+
+```
+Some key points to validate that ingress controller installation succeeded:
+
+- The ingress controller pod is up and running
+- The ingress-nginx service has a external IP address populated
+- If you hit the external IP address of the ingress-nginx, you get the default backend page:
+
+```
+curl http://35.200.150.175
+
+default backend - 404
+```
+
+## Deploying Function with ingress
+
+### Create a function
+
+We will create an environment, function and test that it works:
+
+```
+$ fission env create --name nodejs --image fission/node-env
+environment 'nodejs' created
+
+$ cat hello.js 
+module.exports = async function(context) {
+    return {
+        status: 200,
+        body: "Hello, Fission!\n"
+    };
+}
+
+$ fission fn create --name hello --env nodejs --code hello.js 
+function 'hello' created
+
+$ fission fn test --name hello
+Hello, Fission!
+
+```
+
+### Create a internal route
+
+Let's create a route which is not exposed via the ingress controller:
```

Comment:
Same as previous two comments, explain why we need to do this. What will happen.